### DISCUSSION TITLE
rich table(or chart?) content is not fully extracted

### DISCUSSION THREAD

**Discussion by shaojun** [pymupdf/PyMuPDF#2798] [Looking for help] [answered]
**Is your feature request related to a problem? Please describe.**
I'm trying to develop a `pdf to ppt` tool, and the first step is to parse everything from pdf.

Now I have a `pdf` file has a page with **2** tables(maybe **2** charts?) content in it like:
![image](https://github.com/pymupdf/PyMuPDF/assets/3241829/56e39ba2-c86a-40ba-8090-9088481e83cb)

and the library do can detect table in this page, but just **1**, by use:
```
tabs = page.find_tables()  # locate and extract any tables on page
if tabs.tables:  # at least one table found?
    for tabno, tab in enumerate(tabs, start=1):
        table_extract = tab.extract()  # get the table extracted from the page
        # print(table_extract)  # print content of first table
        pdfPageDescriptor.tables.append(table_extract)
```

the `table_extract` is actually **just one** 2 dimention array of:

![image](https://github.com/pymupdf/PyMuPDF/assets/3241829/e3246412-004f-43d9-bb8e-766affdaa914)

can see both the table or chart structure and data are in mess.

and also, I tried `page.get_images()` on this page, **none of them** related to the tables or charts area.
**Describe the solution you'd like**
parse out the tables or charts.

**Describe alternatives you've considered**
crop the `tables` or `charts`  area as a image is also acceptable.

**Additional context**

**Comment by JorjMcKie** [comment]
There is no possible enhancement here I'm afraid.
This is a typical "Discussions" item, so let me transfer this post first.

**Comment by JorjMcKie** [comment]
We are not looking at a table at all on the shown page. There are vector graphics with some explanatory text - that's all.

You have to analyze this situation yourself, for example by checking if some vector graphic or image is contained in the detected table's bbox.
Vector graphics can be extracted via `paths = page.get_drawings()`. If the table finder thinks it has detected a table, you can check:
```python
paths = page.get_drawings()
tab_bbox = fitz.Rect(tab.bbox)
if [p for p in paths if p["rect"] in tab_bbox] != []:  # some graphics are in table area!
        print("not a table")
```

**Reply by shaojun** [reply]
thanks @JorjMcKie.

your code sample:
```
if [p for p in paths if p["rect"] in tab_bbox] != []:  # some graphics are in table area!
        print("not a table")
```
indicates a suspicious chart area is found at position `tab_bbox `, right? 
then the next step is to extract this area, I believe the most easy way to keep it is to crop this area to a image, could you show me how to crop an area and save it? I just find only thing about crop is targeting a whole page: https://pymupdf.readthedocs.io/en/latest/the-basics.html#cropping-a-pdf

**Reply by JorjMcKie** [reply, marked as answer]
I believe you were taken in the wrong direction by using the verb "crop" which is not applicable here.
What you seem to mean is taking a picture of **_only the drawing_**. This works by using the drawing's bbox as a **_clip_** area like this: `pix = page.get_pixmap(clip=bbox)`. Then do `pix.save("some.jpg")` (or `.png`).
If you want, you can improve image resolution by using the dpi parameter when creating the pixmap.
